test/test_quantized.py

```diff
@@ -1055,109 +1055,121 @@ class TestDynamicQuantizedLinear(TestCase):
         use_bias=st.booleans(),
         use_relu=st.booleans(),
         use_multi_dim_input=st.booleans(),
-        use_channelwise=st.booleans())
+        use_channelwise=st.booleans(),
+        qengine=st.sampled_from(("qnnpack", "fbgemm")))
     def test_qlinear(self, batch_size, input_channels, output_channels,
-                     use_bias, use_relu, use_multi_dim_input, use_channelwise):
-        qlinear_prepack = torch.ops.quantized.linear_prepack
-        if use_relu:
-            qlinear_dynamic = torch.ops.quantized.linear_relu_dynamic
-        else:
-            qlinear_dynamic = torch.ops.quantized.linear_dynamic
-
-        if use_multi_dim_input:
-            batch_size *= 3  # Test the multi-dim input tensor
-
-        X_scale = 1.0
-        X_zp = 0
-        X_value_min = 0
-        X_value_max = 255
-        X_q0 = np.round(np.random.rand(batch_size, input_channels) *
-                        (X_value_max - X_value_min)
-                        + X_value_min
-                        ).astype(np.uint8)
-        X_q0 = np.round(np.random.rand(batch_size, input_channels) *
-                        (X_value_max - X_value_min) + X_value_min).astype(np.uint8)
-        X_q0[0, 0] = X_value_min
-        X_q0[0, 1] = X_value_max
+                     use_bias, use_relu, use_multi_dim_input, use_channelwise, qengine):
 
-        # W_scale = 1.0
-        # W_zp = 0
-        W_scales = np.ones(output_channels)
-        W_zps = np.zeros(output_channels).astype(np.int)
-        W_value_min = -128
-        W_value_max = 127
-        W_q0 = np.round(
-            np.random.rand(output_channels, input_channels)
-            * (W_value_max - W_value_min)
-            + W_value_min
-        ).astype(np.int8)
-        W_q0[0, 0] = W_value_min
-        W_q0[1, 0] = W_value_max
+        if qengine not in torch.backends.quantized.supported_engines:
+            return
+        if qengine == 'qnnpack':
+            if IS_PPC or TEST_WITH_UBSAN or IS_MACOS:
+                return
+            use_channelwise = False
+            use_multi_dim_input = False
+            use_relu = False
```

Comment:
Can we add support for FcRelu to be consistent with server implementation?

Reply:
I'll add it in a subsequent PR. We will need to modify the kernels as well to add this.